### PR TITLE
add orders listing and integration tests for orders

### PR DESCRIPTION
**Pull request by heyitsmaimai**

### CODE REVIEW

File: orders/handlers/create.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const OrdersController = require('../controller');
+const Model = require('../models/OrderModel');
+const collectionHandler = require('../../util/collectionHandler');
+
+module.exports = [
+  collectionHandler(Model),
```
Comment (Wardormeur) [inline orders/handlers/create.js]: Create shoudln't need the collectionHandler : we will never pass down the page/pagesize to this endpoint

File: test/integration/orders.spec.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      'ticketType', 'sessionId', 'dojoId',
+      'ticketId', 'orderId']);
+    expect(res.body.applications[0].id).to.equal('9c56d4ac-7dbc-4b71-897c-15fcb14639d1');
+    expect(res.body.applications[1].id).to.equal('0b9f163e-1705-4bd7-bfbc-a643662b3346');
+    expect(res.body.applications[0].ticketId).to.equal('58544293-9d1e-4ae0-b061-e005225886b2');
+    expect(res.body.applications[1].ticketId).to.equal('58544293-9d1e-4ae0-b061-e005225886b2');
+    expect(res.body.applications[0].orderId).exist;
+    expect(res.body.applications[1].orderId).exist;
+  });
+  it('should return 400 if userId and/or applications and/or eventId is/are not valid', async () => {
```
Comment (Wardormeur) [inline test/integration/orders.spec.js]: All of them are invalid : it doesn't ensure that all of those are individually blocking.
Duplicating the test is probably the easiest

File: test/integration/orders.spec.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      .post('/orders')
+      .send({
+        userId: '1234',
+        applications: [],
+        eventId: '1234',
+      })
+      .set('Accept', 'application/json')
+      .expect(400);
+  });
+  it('should return 400 if any item in an application object is not valid', async () => {
```
Comment (Wardormeur) [inline test/integration/orders.spec.js]: Specify which one ;)
I believe it's status but I can't be sure

**Comment by Wardormeur** [comment]
Requires https://github.com/CoderDojo/cp-events-service/pull/183/files

File: test/integration/orders.spec.js (hunk 2 above)
Comment (heyitsmaimai) [reply test/integration/orders.spec.js]: Gotcha, will do

**Comment by Wardormeur** [comment]
LGTM :+1:  :)

File: orders/controller.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  static async list({ query }, builder = OrderModel.query()) {
+    return builder
+      .allowEager('[applications]')
+      .eager('applications')
+      .modifyEager('applications', _builder =>
+        _builder.applyFilter('awaiting'))
+      .where(query);
+  }
+  static async create({ eventId, userId, applications }) {
+    return OrderModel
```
Comment (Wardormeur) [inline orders/controller.js]: Adding : ensure the status reflects the current status of the event. It shouldn't be passed by the front-end

Comment (Wardormeur) [reply orders/controller.js]: This should be "approved" or "pending" only, depending on the status of the event ('requiresApproval/not)

Comment (Wardormeur) [reply orders/controller.js]: Sorry. On insert is valid :+1:

**Pull request merged**